src/wp-includes/blocks.php

```diff
@@ -1286,22 +1293,44 @@ function insert_hooked_blocks_into_rest_response( $response, $post ) {
 			'ignoredHookedBlocks' => $ignored_hooked_blocks,
 		);
 	}
+
+	if ( 'wp_navigation' === $post->post_type ) {
```

Comment:
What about synced patterns `wp/block`? Should it be extensible? These post types are special do maybe it’s fine as is but a filter here could be useful.

Reply:
Ah, good point. I guess the `wp_block` post type should map to `core/block`.

A filter might make sense here 👍 Something to map post types to block types. (I wonder if we already have a function or data structure to do that somewhere 🤔 )

Reply:
I've finally tested this with a synced pattern, and it's currently _not_ inserting hooked blocks.

Simply adding `elseif ( 'wp_block' === $post->post_type ) { $wrapper_block_type = 'core/block'; }` didn't do the trick, so I'll have to look into this a bit more. However, since this PR has been open for a while and seems to be ready otherwise, I'll merge it and tackle synced patterns in a follow-up 😊

Reply:
That's fair. It was also discussed in https://github.com/WordPress/wordpress-develop/pull/7898/files#r1867470796 to add support for every post type. The challenge with Patterns is that they are handled out of the box if they come from files bundled from the theme or when registered with PHP. The last missing puzzle would be a synced version of the pattern.

Reply:
Update: Support for synced patterns was implemented in https://github.com/WordPress/gutenberg/pull/68058 and https://github.com/WordPress/wordpress-develop/pull/8015, respectively.

I've filed a Trac ticket for other CPTs (and filterable post type -- block type mappings): https://core.trac.wordpress.org/ticket/62715